core/src/main/java/io/crate/types/ObjectType.java

```diff
@@ -70,6 +73,34 @@ public String getName() {
         return (Map<String, Object>) value;
     }
 
+    @Override
+    public Object valueForSets(Object value) throws IllegalArgumentException, ClassCastException {
+        if (value instanceof Map) {
+            Map<String, Object> m = (Map<String, Object>) value;
+            HashMap<String, Object> result = new HashMap<>();
```

Comment:
this loses the order of elements I think - eg. if `value` is a `LinkedHashMap` we're returning a `HashMap` which has different semantics when it comes to ordering. Shall we add some assertions at least?

Reply:
What kind of assertions would you add? Currently we're not using the values - they're just used for contains checks and we keep the original values around as well.

Maybe some javadoc to `valueForSets` to clarify the behaviour?